Security roles in tech

rudri182 opened the pull request.

rudri182 commented:
fixed issue #683. The slide show is remaining and I will on it.

Kundan28 commented:
Please mentioned the issue number you have fixed in this PR. #683 is the PR number instead.

manaswinidas commented on roles-in-tech/security.md:
```diff
@@ -39,9 +49,29 @@ Apprentices will learn secure development basics, common pitfalls, and how to av
+
+- **Cyber Security is just a form of defense** - “The best defense is a good offense.”
+
+- **You will never attacked or breached** - Unfortunately, the attacked business is often the last to know that it's been breached. It's only after banks discover fraudulent charges that businesses hear about a suspected hack.
```
You will never 'be' attacked

manaswinidas commented on roles-in-tech/security.md:
```diff
@@ -39,9 +49,29 @@ Apprentices will learn secure development basics, common pitfalls, and how to av
+
+- **You will never attacked or breached** - Unfortunately, the attacked business is often the last to know that it's been breached. It's only after banks discover fraudulent charges that businesses hear about a suspected hack.
+
+- **Hackers are criminals** - It is a big misconception. There are type of hackers and not all hackers are criminals. There are three types of hackers called - *White Hat Hackers(Who hacks legally)* , *Grey Hat Hackers(who may sometimes violate laws or typical ethical standards, but does not have the malicious intent)* and *Black Hat Hackers(Illegal hacking)*.
```
Hackers(who 'hack' legally) and Grey Hat Hackers (..... but do not have)

manaswinidas commented on roles-in-tech/security.md:
```diff
@@ -39,9 +49,29 @@ Apprentices will learn secure development basics, common pitfalls, and how to av
+
+- **Hackers are criminals** - It is a big misconception. There are type of hackers and not all hackers are criminals. There are three types of hackers called - *White Hat Hackers(Who hacks legally)* , *Grey Hat Hackers(who may sometimes violate laws or typical ethical standards, but does not have the malicious intent)* and *Black Hat Hackers(Illegal hacking)*.
+
+- **If wi-fi has a passwords, then it is secure** - If the person who travels to different workplaces and need to use public wi-fi it is not recommended to do sensitive transactions through it. The same password is shared with many users and anyone can see the information that is being transferred.
```
Travels to different workplaces and 'needs'

manaswinidas commented on roles-in-tech/security.md:
```diff
@@ -39,9 +49,29 @@ Apprentices will learn secure development basics, common pitfalls, and how to av
+
+- **If wi-fi has a passwords, then it is secure** - If the person who travels to different workplaces and need to use public wi-fi it is not recommended to do sensitive transactions through it. The same password is shared with many users and anyone can see the information that is being transferred.
+
+- **By turning of locations you will not be traced** - Almost half of people surveyed either didn't know or were unsure whether disabling GPS prevents all tracking. Your phone's built-in location service is just one way data can be collected. As Pew Research notes, your phone also connects to cell towers and Wi-Fi networks, and connection to these technologies allows tracking too.
```
Turning 'off' and I think GPS will be more appropriate

manaswinidas commented on roles-in-tech/security.md:
```diff
@@ -39,9 +49,29 @@ Apprentices will learn secure development basics, common pitfalls, and how to av
+
+- **By turning of locations you will not be traced** - Almost half of people surveyed either didn't know or were unsure whether disabling GPS prevents all tracking. Your phone's built-in location service is just one way data can be collected. As Pew Research notes, your phone also connects to cell towers and Wi-Fi networks, and connection to these technologies allows tracking too.
+
+- **Emails are secure** - Over 400 businesses are targeted daily, and these emails are growing increasingly challenging to identify since they’re designed to mimic legitimate communication so they can gain access to your sensitive information – including usernames and passwords.
```
growing increasingly and it is challenging

manaswinidas commented on roles-in-tech/security.md:
```diff
@@ -1,12 +1,20 @@
+
+![security locks](\roles-in-tech\security-locks.jpg)
+
+
```
Remove extra line , .i.e., line 6

Kundan28 commented:
@manaswinidas Would you please tell me which issue this PR solves ? So that I can add it to the score sheet.

manaswinidas commented:
@Kundan28  I suppose it is #648. @rudri182 Please resolve the conflicts.

rudri182 commented:
I will resolve it by today.

On Thu, Mar 21, 2019, 1:17 AM Manaswini Das <notifications@github.com>
wrote:

> @Kundan28 <https://github.com/Kundan28> I suppose it is #648
> <https://github.com/Techtonica/curriculum/issues/648>. @rudri182
> <https://github.com/rudri182> Please resolve the conflicts.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/Techtonica/curriculum/pull/683#issuecomment-475000731>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AeLQ1UXqYoEbRFuqN8pr0ZHnjHS41UT6ks5vYpBngaJpZM4b_b66>
> .
>

alodahl commented on roles-in-tech/security.md:
```diff
@@ -1,12 +1,19 @@
 90-180 minutes
 
 ### Prerequisites
 How the Internet Works Topic Outline
```
Can you please also add the “intro to security” outline here?

Please check out the earlier intro to security outline and make sure to remove any copied sections (this roles outline was originally just a copy of intro to security) unless the lines are also especially useful for the roles topic.

rudri182 replied:
Yes, I will definitely remove it. When I read the file I saw that it is copied, but I thought that they are approved previously so I didn't remove it.

alodahl replied:
Thank you! 😄

alodahl commented:
I saw you started working on the slideshow, @rudri182 !  Please be sure to follow the formatting guidelines found in #600 .

rudri182 commented:
Yes sure. I will 😊

On Mon, Mar 25, 2019, 2:24 AM Alina L. <notifications@github.com> wrote:

> I saw you started working on the slideshow, @rudri182
> <https://github.com/rudri182> ! Please be sure to follow the formatting
> guidelines found in #600
> <https://github.com/Techtonica/curriculum/issues/600> .
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/Techtonica/curriculum/pull/683#issuecomment-475998970>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AeLQ1cWl02KniZ2Mzrr8Hg6dZIU9xuyNks5vZ-XpgaJpZM4b_b66>
> .
>

alodahl commented on roles-in-tech/security.md:
```diff
@@ -1,13 +1,14 @@
 
 ### Prerequisites
-How the Internet Works Topic Outline
+- To become security engineer, one should pursue bachelors degree in computer or IT field or have experience in security field.
```
If people reading this are reading because they are trying to take this free bootcamp, is there another way they could become a security engineer?

rudri182 replied:
While preparing for this role I came across this in many sites so I added it. I have changed it in the file, t is not necessary but it is recommended to have at least a bachelors degree in computer science or IT field. And also some knowledge of networking and programming is beneficial for a security engineer, therefore they may require a bachelors degree. Here are some links that I referred to:
[BECOME A SECURITY ENGINEER](https://www.cyberdegrees.org/jobs/security-engineer/)
[HOW DO I BECOME A CYBERSECURITY ENGINEER?](https://www.onlineengineeringprograms.com/faq/how-do-i-become-a-cybersecurity-engineer)

If there are any required changes, I am happy to solve it.

alodahl commented on roles-in-tech/security.md:
```diff
@@ -17,35 +18,55 @@ How the Internet Works Topic Outline
+
+### Roles and Responsibilities
+
+- developing a set of security standards and practices
```
could you change these to "develop", "create", etc, rather than using -ing words?

alodahl commented on roles-in-tech/security.md:
```diff
@@ -68,33 +89,26 @@ These are some misconceptions that people have in their mind but it is not alway
-  - Also consider [Auth0](/oauth/o-auth.md).
-- Next, you can validate user input with Parsley and validate fields with Validator.
-- Use DOMPurify to prevent XSS.
+- Talk to as many professionals as you can [National Cyber Security Centre](https://live.ncsc.gov.uk/topics/new-talent) is worth for who lives in UK
```
This line is pretty strange... please look at it in the markdown preview and change it.

alodahl commented on roles-in-tech/security.md:
```diff
@@ -68,33 +89,26 @@ These are some misconceptions that people have in their mind but it is not alway
-
-The next step is to try to XSS each other's sites.
+- You can start playing [CTFs](https://ctftime.org/ctf-wtf/). It is a great way to learn ethical hacking.
+- Learn how to code! - [codecademy](https://www.codecademy.com/about),
```
Line 105 and 106 are a little too general - could you remove them or replace them with something about role sin security?

rudri182 commented:
I have modified the slideshow as per the comments and also added some content. Please review it and recommend me for more slides or content if it is required.

alodahl commented on roles-in-tech/security.md:
```diff
@@ -99,10 +101,8 @@ These are some misconceptions that people have in their mind but it is not alway
 
-
 ### Check for Understanding
-- You can start playing [CTFs](https://ctftime.org/ctf-wtf/). It is a great way to learn ethical hacking.
```
Wait!  Can you please keep the CTFs one and remove the others? 😅

alodahl commented on roles-in-tech/security.md:
```diff
@@ -90,7 +90,9 @@ These are some misconceptions that people have in their mind but it is not alway
-- Talk to as many professionals as you can [National Cyber Security Centre](https://live.ncsc.gov.uk/topics/new-talent) is worth for who lives in UK
+- *Talk to as many professionals as you can*
+
+ [National Cyber Security Centre](https://live.ncsc.gov.uk/topics/new-talent) is worth for who lives in UK
```
can you describe this link a little more here?  All I know by looking at it is that it probably happens in the UK.

alodahl commented on roles-in-tech/security.md:
```diff
@@ -48,25 +48,25 @@ After gaining basic skills one can go further as *Security Architect* or *Securi
-- helping plan an organization’s information security strategy
-- educating staff members on information security through training and awareness
-- recommending modifications in legal, technical and regulatory areas
+- develops a set of security standards and practices
```
Change these to command words - develop, create, recommend...

alodahl commented on roles-in-tech/security.md:
```diff
@@ -1,71 +1,114 @@
 
 - [Techtonica's Roles in Tech Presentation](https://drive.google.com/file/d/1oXgOX1EC4igI6ZwgJlhF6AN4dG2Yn_S4/view) (security section starts at 2:57)
-- [A quick introduction to web security [FreeCodeCamp]](https://medium.freecodecamp.org/a-quick-introduction-to-web-security-f90beaf4dd41)
+- online course at [CYBRAY](https://www.cybrary.it/)
```
Great resources, but are you assigning all of these to be done during the 30-45 window?  If they're not assigned, maybe you can put them in a new category like you have for "conferences" and "organizations."

alodahl commented on roles-in-tech/security.md:
```diff
@@ -1,71 +1,114 @@
-The next step is to try to XSS each other's sites.
+- [27 ways to learn how to code on the cheap (or free)](https://thenextweb.com/dd/2017/04/03/so-you-want-to-be-a-programmer-huh-heres-25-ways-to-learn-online/#.tnw_hZDm58Mu)
 
 ## Understanding the Field
```
Move "Understanding the field" and "What kinds of security jobs" sections to before "guided practice".

alodahl commented:
The slideshow looks great , @rudri182 👍🏼

rudri182 commented:
I have made changes in the markdown file. What is next to be done?  Some more content to add in slideshow?

alodahl commented:
I think this loos complete; just make sure to resolve the conflicts so we can merge, @rudri182 . @manaswinidas , I'll wait for you to approve.

alodahl commented:
@rudri182 , I got some feedback from some friends about what employers are looking for. Maybe you can add this to your CS Bachelor's Degree comments, and put them in their own section called "What Employers are looking for"

Michelle   [1 day ago]
You'd have to understand things like XSS, CSRF, sql injections, input validation, etc etc

Kerrie   [1 day ago]
I guess it would depend on where you want to go (i.e. pentester, defense, red teaming, compliance, etc…). I was considering going into the more technical route but ended up in security compliance which I love so much more. There are so many routes that I can’t really pin down one but just know that either self taught, bootcamp, or college course, either one will work. It comes down to if you can show what you know and explain it to interviewers. I know for a fact that the info security industry has a huge hiring gap AND they are looking to hire women. You don’t have to know everything but having the attitude that you can learn it and do it says so much more. Let me know if that helps.

Kerrie   [1 day ago]
also, connect with other women security orgs like Women in Security and Privacy, Women Cyberjutsu and other orgs is very helpful. They tend to have free trainings and offer scholarships. Right now you can sign up to get a free scholarship to Def Con.  See if you can connect with their members and find a mentor. That will help you along a lot.

alodahl commented on roles-in-tech/security.md:
```diff
@@ -17,16 +19,22 @@
 
+### What employers are looking for
+
+- You'd have to understand things like XSS, CSRF, SQL injections, input validation, etc.
```
Please change this section's tone to be less conversational/ more like the rest of your outline. Also, avoid using all caps and consider *italics* or **bold** instead.

alodahl replied:
Also, remember to resolve your merge conflicts once this is ready.

rudri182 commented:
I have solved the conflicting part of the file. Please review it. What is the next task to be done?

rudri182 commented:
I have completed the required changes but it may not be reviewed. Please review it and let me know if there are any further changes that is to be done or it is ready to merge.

alodahl commented:
Hi @rudri182 , thanks for the changes!  It still says there are merge conflicts, however - please resolve those so we can merge.
![image](https://user-images.githubusercontent.com/29131689/55566530-ba385e80-56b0-11e9-9831-4900f37dcb20.png)

rudri182 commented:
Thank you for your wonderful response! I am happy to contribute more to **Techtonica/curriculum**. Thank you for quick and continuous reply and waiting for correction from my side and guiding me to resolve conflicts. Excited to solve new issue!

Kundan28 commented on roles-in-tech/security.md:
```diff
@@ -1,94 +1,164 @@
+- [Intro to Security](https://github.com/Techtonica/curriculum/blob/master/security/web-security.md)
 
 ### Motivation
+- Computer security is important, primarily to *keep your information protected*. It’s also important for your computer’s overall health, helping to prevent viruses and malware and allowing programs to run more smoothly.
```
*malwares

rudri182 replied:
I have changed the file but I am facing problem in pushing changes in the merged branch. How can I push committed message?

alodahl replied:
It is already merged. But don’t worry about it; I think malware is already plural, and is correct.

Kundan28 replied:
Okay.
I though if it's written viruses then it should be malwares too.

alodahl replied:
If you have 2 instances of software, do you now have softwares?

Kundan28 replied:
Okay, I got it and by the way virus is also a kind of software.

rudri182 commented:
Cited the text and added the links. @alodahl

alodahl commented:
@rudri182 are you saying you just added more?  If so, where are they? If they are on this branch, you would have to make a new PR since this is already merged.

rudri182 commented:
No, I have not added more. I have put references at the bottom of the
outline for which I have copied . Do I need to make a new PR ?

On Sat, Jun 8, 2019, 8:04 PM Alina L. <notifications@github.com> wrote:

> @rudri182 <https://github.com/rudri182> are you saying you just added
> more? If so, where are they? If they are on this branch, you would have to
> make a new PR since this is already merged.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/Techtonica/curriculum/pull/683?email_source=notifications&email_token=AHRNBVMFBKFGMDGS6U56QJDPZO7QJA5CNFSM4G75X25KYY3PNVWWK3TUL52HS4DFVREXG43VMVBW63LNMVXHJKTDN5WW2ZLOORPWSZGODXHVYOQ#issuecomment-500128826>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AHRNBVJTDOPBZUHU4XNPMEDPZO7QJANCNFSM4G75X25A>
> .
>

alodahl commented:
Yes , you need to make a new PR.

alodahl commented:
Ok, thanks for double-checking!
